Library.Api/Program.cs

```diff
@@ -0,0 +1,54 @@
+using AutoMapper;
+using Library.Application.Dtos;
+using Library.Application.Services;
+using Library.Infrastructure.Repositories;
+using Library.Application;
+using Library.Application.Interfaces;
+
+var builder = WebApplication.CreateBuilder(args);
+
+var mapperConfig = new MapperConfiguration(
+    config => config.AddProfile(new MappingProfile()),
+    LoggerFactory.Create(builder => builder.AddConsole()));
+IMapper? mapper = mapperConfig.CreateMapper();
+builder.Services.AddSingleton(mapper);
+
+builder.Services.AddControllers();
+builder.Services.AddEndpointsApiExplorer();
+builder.Services.AddSwaggerGen();
+
+builder.Services.AddSingleton<BookRepository, BookRepository>();
```

Comment:
Тут как раз и происходит регистрация зависимостей. Мы должны указать сначала интерфейс, а затем реализацию.
Таким образом программа будет понимать что именно должно реализовывать интерфейс.
```suggestion
builder.Services.AddSingleton<IRepository<Book, int>, BookRepository>();
```
Исправить тут и ниже.